include/golioth/location/wifi.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2024 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <stdint.h>
+
+#include <golioth/client.h>
+#include <golioth/golioth_status.h>
+#include <golioth/location.h>
+#include <zcbor_encode.h>
+
+/** @brief Wi-Fi scan result, which is passed to golioth_location_wifi_append() */
+struct golioth_wifi_scan_result
+{
+    /** RSSI */
+    int8_t rssi;
+    /** BSSID */
+    uint8_t mac[6];
+};
+
+/** @brief Wi-Fi location request, used with golioth_location_wifi_append() */
+struct golioth_location_wifi_req
+{
+    uint8_t buf[(16 + (CONFIG_GOLIOTH_LOCATION_WIFI_COUNT_MAX * 30))];
+    zcbor_state_t zse[2 + 1];
+    enum golioth_status status;
+
+    struct golioth_location_rsp *rsp;
+};
+
+void golioth_location_wifi_init(struct golioth_location_wifi_req *req);
+enum golioth_status golioth_location_wifi_append(struct golioth_location_wifi_req *req,
+                                                 const struct golioth_wifi_scan_result *entry);
+enum golioth_status golioth_location_wifi_finish(struct golioth_location_wifi_req *req);
+
+enum golioth_status golioth_location_wifi_get_sync(struct golioth_client *client,
+                                                   struct golioth_location_wifi_req *req,
+                                                   struct golioth_location_rsp *rsp);
```

Comment:
For devices that support both Wi-Fi and cellular we will allow for both types of network data to be included in the same request. I could see value in having dedicated functions for each, but we could also have a single `golioth_location_get_sync` that accepted optional Wi-Fi and cellular requests. Doing so would make it harder enable / disable individually though. What do you think?

Reply:
Makes sense! I somehow missed the case with both technologies used at once. I'll rework current implementation to account for that.